sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -218,7 +219,8 @@ abstract class Optimizer(sessionCatalog: SessionCatalog)
       RewriteCorrelatedScalarSubquery.ruleName ::
       RewritePredicateSubquery.ruleName ::
       PullOutPythonUDFInJoinCondition.ruleName ::
-      NormalizeFloatingNumbers.ruleName :: Nil
+      NormalizeFloatingNumbers.ruleName :: Nil ++
+      SQLConf.get.getConf(OPTIMIZER_NON_EXCLUDABLE_RULES).toSeq.flatMap(Utils.stringToSeq)
```

Comment:
Does it make sense to replace type of `nonExcludableRules` to `Set` since the main method applied to the data structure is `contains`?

Reply:
Yep. It's possible~